ignite/contrib/handlers/time_profilers.py

```diff
@@ -437,3 +438,395 @@ def odict_to_str(d: Mapping):
+
+    def _as_first_started(self, engine):
+        if hasattr(engine.state.dataloader, "__len__"):
+            num_iters_per_epoch = len(engine.state.dataloader)
+        else:
+            num_iters_per_epoch = engine.state.epoch_length
+
+        self.max_epochs = engine.state.max_epochs
+        self.total_num_iters = self.max_epochs * num_iters_per_epoch
+
+        self.event_handlers_names = {
+            e: [
+                h.__qualname__ if hasattr(h, "__qualname__") else h.__class__.__name__
+                for (h, _, _) in engine._event_handlers[e]
+                if not self._is_internal_handler(h)
+            ]
+            for e in self._events
+        }
+
+        self._reset(self.max_epochs, self.total_num_iters)
+
+        self._orig_event_handlers = engine._event_handlers.copy()
+
+        for e in self._global_events:
+            for i, h in enumerate(self._orig_event_handlers[e]):
+                if self._is_internal_handler(h):
+                    pass
+                engine._event_handlers[e][i] = (partial(self._timeit_global_events, h), (e,), {})
+
+        for e in self._epoch_events:
+            for i, h in enumerate(self._orig_event_handlers[e]):
+                if self._is_internal_handler(h):
+                    pass
+                engine._event_handlers[e][i] = (partial(self._timeit_epoch_events, h), (engine, e,), {})
+
+        for e in self._iteration_events:
+            for i, h in enumerate(self._orig_event_handlers[e]):
+                if self._is_internal_handler(h):
+                    pass
+                engine._event_handlers[e][i] = (partial(self._timeit_iteration_events, h), (engine, e,), {})
+
+        # processing timer
+        engine._event_handlers[Events.ITERATION_STARTED].append((self._processing_timer.reset, (), {}))
+        engine._event_handlers[Events.ITERATION_COMPLETED].insert(0, (self._timeit_processing, (engine,), {}))
+
+        # dataflow timer
+        engine._event_handlers[Events.GET_BATCH_STARTED].append((self._dataflow_timer.reset, (), {}))
+        engine._event_handlers[Events.GET_BATCH_COMPLETED].insert(0, (self._timeit_dataflow, (engine,), {}))
+
+        # revert back the wrapped handlers with original handlers at the end
+        engine._event_handlers[Events.COMPLETED].append((self._detach_profiler_handlers, (engine,), {}))
+
+    def attach(self, engine: Engine):
+        if not isinstance(engine, Engine):
+            raise TypeError("Argument engine should be ignite.engine.Engine, " "but given {}".format(type(engine)))
+
+        if not engine.has_event_handler(self._as_first_started):
+            engine._event_handlers[Events.STARTED].insert(0, (self._as_first_started, (engine,), {}))
+
+    def get_results(self):
+        """
+        Method to fetch the aggregated profiler results after the engine is run
+
+        .. code-block:: python
+
+            results = profiler.get_results()
+
+        """
+        total_eh_time = round(
```

Comment:
This black formatting is ugly :) could you please split it into several expressions.